### PR TITLE
Update compiler-rt to LLVM 15

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h [hunk 1]
```diff
@@ -14,11 +14,29 @@
...
+#if !__DARWIN_ONLY_64_BIT_INO_T
+#define SANITIZER_HAS_STAT64 1
+#define SANITIZER_HAS_STATFS64 1
+#else
+#define SANITIZER_HAS_STAT64 0
+#define SANITIZER_HAS_STATFS64 0
+#endif
+#elif SANITIZER_EMSCRIPTEN
+#define SANITIZER_HAS_STAT64 0
+#define SANITIZER_HAS_STATFS64 0
```
Comment (kripken) [inline system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h:33]: I think we do support these. Though maybe we don't want to use them here for some reason?

**Review by kripken — COMMENTED** [review]
Please update the changelog too.

File: system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h (hunk 1 above)
Comment (sbc100) [reply system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h:33]: By default it looks like we don't define those:

```
ache/sysroot/lib/wasm32-emscripten/libsanitizer_common_rt.a" for subsequent builds)
/usr/local/google/home/sbc/dev/wasm/emscripten/system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp:214:33: error: invalid application of 'sizeof' to an incomplete type 'struct statfs64'
  unsigned struct_statfs64_sz = sizeof(struct statfs64);
                                ^     ~~~~~~~~~~~~~~~~~
```

We could define _GNU_SOURCE or _LARGEFILE_SUPPORT but I don't think any of that is necessary.. we didn't previously have SANITIZER_INTERCEPT_STATFS64 or SANITIZER_INTERCEPT_STAT64 defined and none of our tests are failing.

Comment (kripken) [reply system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h:33]: Interesting... maybe they're behind a flag or something then. Or maybe I'm misremembering...

**Comment by sbc100** [comment]
I'm gonna wait for the next emsdk release before landing any of the llvm 15 updates.

**Pull request merged**